core/state_transition.go

```diff
@@ -269,6 +269,13 @@ func (st *stateTransition) buyGas() error {
 			mgval = mgval.Add(mgval, l1Cost)
 		}
 	}
+	var operatorCost *big.Int
```

Comment:
I am guessing the test coverage for this is in the monorepo?

Reply:
I'll have a look, and add it if not.

Reply:
Added there: https://github.com/ethereum-optimism/optimism/pull/12166/commits/19a01efaa67dd5af950e19e50f7b97be7c368724

Reply:
Does this test coverage run as part of the CI on this PR? Would be nice to have those guardrails in place.

Reply:
@teddyknox I don't think it does. We can run the monorepo CI on the partner PR here https://github.com/ethereum-optimism/optimism/pull/12166/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6 (but probably need to update the reference / tag to run the latest version of this PR). @leruaa does this make sense to you / can we do that step now?

Reply:
It would be great to try to parameterize some existing upstream geth tests of the STF to also cover the case with enabled operator fees. We did this for other OP-Stack features too. So we don't need to reinvent a test from scratch but just take existing upstream tests and run the twice with and without operator fee enabled.

Reply:
Is this the kind of test you're looking for? https://github.com/ethereum-optimism/optimism/pull/12166/files#diff-d33cb27fc349b5e4d8dc2fbb74661aa10c55d2cbea45a3a42965c53a6381db86

Reply:
I meant also adding a (or extending an existing) test to op-geth itself, so we don't only rely on the monorepo testing this. I think such a critical implementation that changes gas accounting should have a test inside op-geth as well. Monorepo tests only run on new op-geth code when the op-geth dependency is updated.

Reply:
Tracking here https://github.com/ethereum-optimism/op-geth/issues/528